### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by jessedoyle**
### Usage

```ruby
class HelloWorldHandler < Aws::Lex::Conversation::Handler::Base
  def will_respond?(conversation)
    true # respond to any input event
  end

  def response(conversation)
    # simply close the conversation with a "Hello, World" response.
    conversation.close(
      fulfillment_state: 'Fulfilled',
      message: {
        content: 'Hello, World!'
      }
    )
  end
end

def my_lambda_handler(event:, context:)
  conversation = Aws::Lex::Conversation.new(event: event, context: context)
  conversation.handlers = [{
    handler: HelloWorldHandler
  }]
  conversation.respond
end
```

### TODO 
- [x] Full Test Coverage
- [x] Update README

### CODE REVIEW

File: lib/aws/lex/conversation.rb [hunk 1]
```diff
@@ -1,12 +1,54 @@
...
+      end
+
+      def handlers
+        chain.map(&:class)
+      end
+
+      def respond
+        chain.first.handle(self)
+      end
+
```
Comment (mvandenbeuken) [inline lib/aws/lex/conversation.rb:44]: is it worth making an intent_name convenience method, or current_intent?

File: lib/aws/lex/conversation/support/inflector.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      module Support
+        class Inflector
+          attr_accessor :input
+
+          def initialize(input)
+            self.input = input.to_s
+          end
+
+          def to_snake_case
+            # shamelessly borrowed from ActiveSupport
```
Comment (mvandenbeuken) [inline lib/aws/lex/conversation/support/inflector.rb:15]: 😄

File: lib/aws/lex/conversation.rb (hunk 1 above)
Comment (jessedoyle) [reply lib/aws/lex/conversation.rb:44]: Yes, that's probably a good thing! I'll add one!

Comment (jessedoyle) [reply lib/aws/lex/conversation.rb:44]: I've added `Conversation#intent_name` as a convenience method!

Comment (mvandenbeuken) [reply lib/aws/lex/conversation.rb:44]: Good stuff!

File: lib/aws/lex/conversation/version.rb [hunk 3]
```diff
@@ -2,8 +2,8 @@
 
 module Aws
   module Lex
-    module Conversation
-      VERSION = '0.1.0'
+    class Conversation
+      VERSION = '0.5.0'
```
Comment (mvandenbeuken) [inline lib/aws/lex/conversation/version.rb:6]: good call!

**Pull request merged**